nucleus/v1/types.go

```diff
@@ -0,0 +1,187 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+
+// NucleusHub represents a deployment of nucleus hub component.
+type NucleusHub struct {
```

Comment:
Let's finalize the name before we merge this PR. 'nucleus' is a term I found convenient to illustrate the concept but not a name I would want in production.

Some ideas:

- ControlPlane
- Hub
- HubCore
- ControlPlaneCore

When we think about the name, think about something that would be good to layer product onto in a different resource(s), rather than a place where we will grow a lot of knobs for product. The intent should be that this resource will stay small and deal only with the most essential parts of the system, and contain no product decisions.